services/provider/proto/provider.proto

```diff
@@ -153,6 +153,8 @@ message StorageClaimConfigRequest{
 message StorageClaimConfigResponse{
     // ExternalResource holds the configuration data of external storage cluster
     repeated ExternalResource externalResource = 1;
+    // MetaData contains additional info regarding the external cluster resources
+    bytes MetaData = 2;
```

Comment:
after going through PR, why not `bool enableMirroring = 2;`? why do we need to convert bool (`cbp.Spec.Mirroring.Enabled`) into string (`%v`) and then encode to bytes (`json.Marshal`)

The default proto value will be `false` anyways, unless `false` is also a valid value like after enabling you could disable we don't need to use bool pointer.

Reply:
See first link in the description of the attached Jira for design details, @leelavg. We're looking at a feature that would allow us to add _multiple_ fields that are considered as metadata. Once this PR's WIP tag is removed, there will be two flags appended to the metadata field.